tools/protocols/src/Server.ts

```diff
@@ -132,21 +159,168 @@ export class Server {
+    // clientOnly flag set when initializing client for last look `senderServer`
+    const parsedUrl = parseUrl(locator)
+    const options = {
+      protocol: parsedUrl.protocol,
+      hostname: parsedUrl.hostname,
+      port: parsedUrl.port,
+      timeout: REQUEST_TIMEOUT,
+    }
+
+    if (!clientOnly) {
+      this.supportedProtocols = [
+        { name: 'request-for-quote', version: '2.0.0' },
+      ]
+      this.isInitialized = true
+    }
+
+    if (isBrowser) {
+      const jaysonClient = require('jayson/lib/client/browser')
+      this.httpClient = new jaysonClient((request, callback) => {
+        fetch(url.format(parsedUrl), {
+          method: 'POST',
+          body: request,
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        })
+          .then((res) => {
+            return res.text()
+          })
+          .then((text) => {
+            callback(null, text)
+          })
+          .catch((err) => {
+            callback(err)
+          })
+      }, options)
+    } else {
+      const jaysonClient = require('jayson/lib/client')
+      if (options.protocol === 'https:') {
+        this.httpClient = jaysonClient.https(options)
+      } else {
+        this.httpClient = jaysonClient.http(options)
+      }
+    }
+  }
+
+  private async _initWebSocketClient(locator: string) {
+    this.webSocketClient = new JsonRpcWebsocket(
+      url.format(locator),
+      REQUEST_TIMEOUT,
+      (error: JsonRpcError) => {
+        this.emit('error', error)
+      }
+    )
+    const initPromise = new Promise<SupportedProtocolInfo[]>(
+      (resolve, reject) => {
+        setTimeout(() => {
+          reject('Server did not call initialize in time')
+          this.disconnect()
+        }, REQUEST_TIMEOUT)
+
+        this.webSocketClient.on('initialize', (message) => {
+          this.initialize(message)
+          this.isInitialized = true
+          resolve(this.supportedProtocols)
+          return true
+        })
+      }
+    )
+
+    this.webSocketClient.on('updatePricing', this.updatePricing.bind(this))
+    await this.webSocketClient.open()
+    await initPromise
+  }
+
+  private requireInitialized() {
+    if (!this.isInitialized) throw new Error('Server not yet initialized')
+  }
+
+  /**
+   * Throws if RFQ protocol is not supported, or if server is not yet
+   * initialized.
+   */
+  private requireRFQSupport(version?: string) {
+    if (!this.supportsProtocol('request-for-quote', version))
+      throw new Error(
+        `Server at ${this.locator} doesn't support this version of RFQ`
+      )
+  }
+
+  /**
+   * Throws if Last Look protocol is not supported, or if server is not yet
+   * initialized.
+   */
+  private requireLastLookSupport(version?: string) {
+    if (!this.supportsProtocol('last-look', version))
+      throw new Error(
+        `Server at ${this.locator} doesn't support this version of Last Look`
```

Comment:
Possible to conditionally throw either "Server at ... doesn't support Last Look" or if version is set "Server at ... doesn't support this version of Last Look"?